[WIP] Shorten and improve CONTRIBUTING.md #10892

cecilialiao opened the pull request:
Fixes #10892, by simplifying `CONTRIBUTING.md` and referencing to `doc\` where appropriate.

cecilialiao committed:
Reuse intro from contributing.rst

cecilialiao committed:
Removed content already on rst

Simplified CONTRIBUTING.md by removing content that already
exists in contributing.rst. Added Code of Conduct reference
as seen in a few other CONTRIBUTING.md of popular libraries.

cecilialiao committed:
Fixed markdown formatting

qinhanmin2014 commented:
I think we need to ensure that everything we remove in CONTRIBUTING.md is included the contributing guide. E.g., some entries in Pull Request Checklist are not included in the contributing guide.

cecilialiao commented:
That's a great shout, I will look to add the deleted content from `CONTRIBUTING.MD` to `contributing.rst` if it's not already on the latter. Can I check on best practice on git usage? Should I make those changes to `contributing.rst` in the same branch `shorten-contributing-md`?

qinhanmin2014 commented:
Yes, simply push commits to that branch and they'll appear in the PR

cecilialiao committed:
Update contributing.rst

In shortening `CONTRIBUTING.md`, material deleted from `how to contribute` section are added onto `contributing.rst` if it doens't exist on the latter.

jnothman commented on CONTRIBUTING.md:
```diff
@@ -2,239 +2,41 @@
+
+But there are many other ways to help. In particular answering queries on the
+[issue tracker](https://github.com/scikit-learn/scikit-learn/issues),
+investigating bugs, and [reviewing other developers pull 
```
Apostrophe after "developers" please

jnothman commented on CONTRIBUTING.md:
```diff
@@ -2,239 +2,41 @@
+Another way to contribute is to report issues you're facing, and give a "thumbs
+up" on issues that others reported and that are relevant to you. It also helps
+us if you spread the word: reference the project from your blog and articles,
+link to it from your website, or simply star to say "I use it". 
```
"star" -> "star it in GitHub"

jnothman reviewed (commented):
This looks much better IMO

eharris99 commented:
Hi I also submitted some proposed edits to this file
